webpack/assets/javascripts/react_app/routes/index.js

```diff
@@ -1,26 +1,38 @@
 import React from 'react';
-import { Switch, Route } from 'react-router-dom';
+import { Switch } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
 import { routes } from './routes';
+import LayoutRoute from './RouteWithLayout';
+import { layoutPropTypes } from '../components/Layout/Layout';
+import ForemanRoute from './ForemanRoute';
 
 let currentLocation = null;
 
-const AppSwitcher = () => (
+const removeRootContainer = renderProps => {
+  const railsContainer = document.getElementById('rails-app-content');
+
+  if (railsContainer) {
+    railsContainer.remove();
+  }
+  currentLocation = renderProps.location;
+};
+
+const AppSwitcher = ({ data: { layout } }) => (
   <Switch>
-    {routes.map(({ render, path, ...routeProps }) => (
-      <Route
-        path={path}
+    {routes.map(({ render, path, skipLayout, ...routeProps }) => (
+      <ForemanRoute
         key={path}
+        render={render}
+        layout={layout}
```

Comment:
To have a configurable layout, should each route specify `layout` rather than all routes having the same value from prop?